params/json.libevm.go

```diff
@@ -124,3 +124,52 @@ func toJSONRawMessages(v any) (map[string]json.RawMessage, error) {
 	}
 	return msgs, nil
 }
+
+// UnmarshalChainConfigJSON JSON decodes the given data into the config and the extra
+// pointer arguments. The root JSON keys except the "extra" key are decoded into
+// the config argument, and the object at the "extra" key, if present, is decoded
+// into the extra argument. Note the extra argument must be a non-nil pointer or
+// an error would be returned.
+func UnmarshalChainConfigJSON[T any](data []byte, config *ChainConfig, extra *T) (err error) {
+	switch {
+	case config == nil && extra == nil:
+		return fmt.Errorf("chain config and extra config are both nil")
+	case extra == nil:
+		// non-registered extra call from ChainConfig.UnmarshalJSON
+		// we only want to decode to the chain config, ignoring the
+		// "extra" JSON key.
+		err = json.Unmarshal(data, (*chainConfigWithoutMethods)(config))
+		if err != nil {
+			return fmt.Errorf("decoding chain config without extra: %s", err)
+		}
+		return nil
+	case config == nil:
+		// decode the data to the extra argument only.
+		// this originates from the registered extra + re-use JSON
+		// root call from ChainConfig.UnmarshalJSON.
+		err = json.Unmarshal(data, extra)
+		if err != nil {
+			extra = nil
+			return fmt.Errorf("decoding chain config to %T: %s", extra, err)
+		}
+		return nil
+	default:
+		// Decode the data separately to the chain config and the extra.
+		err = json.Unmarshal(data, (*chainConfigWithoutMethods)(config))
+		if err != nil {
+			return fmt.Errorf("decoding root chain config: %s", err)
+		}
+
+		jsonExtra := struct {
+			Extra *T `json:"extra"`
+		}{
+			Extra: extra,
+		}
```

Comment:
we should support the "ReuseJSONRoot" option here like we do here: https://github.com/ava-labs/go-ethereum/blob/44f23c88698df4567d2addede829e669f09f5618/params/json.libevm.go#L43-L59

This is because in the C-Chain genesis and the genesis of the existing subnets the "extra" configurations are directly included in the root json object and are not nested under `"extra": {...}`

Reply:
Done I hope 😉